shared/constants/types/fs.js

```diff
@@ -2,21 +2,40 @@
 import * as I from 'immutable'
 
 export opaque type Path = ?string
-export type FolderPathType = 'folder'
-export type FilePathType = 'file' | 'symlink' | 'exec'
-export opaque type PathType = FolderPathType | FilePathType
-export type _FilePathItem = {
-  type: FilePathType,
-}
+
+export type PathType = 'folder' | 'file' | 'symlink' | 'exec' | 'unknown'
+
 export type _FolderPathItem = {
-  type: FolderPathType,
+  type: 'folder',
   children: I.List<string>,
 }
-export type PathItem = I.RecordOf<_FolderPathItem> | I.RecordOf<_FilePathItem>
-export type PathItems = I.Map<Path, PathItem>
+export type FolderPathItem = I.RecordOf<_FolderPathItem>
+
+export type _SymlinkPathItem = {
+  type: 'symlink',
+  linkTarget: Path,
+}
+export type SymlinkPathItem = I.RecordOf<_SymlinkPathItem>
+
+export type _FilePathItem = {
+  type: 'file',
+}
+export type FilePathItem = I.RecordOf<_FilePathItem>
+
+export type _ExecPathItem = {
```

Comment:
its not clear to me what makes exec special. or if it is based on types then file is just a generic file and we'd have types for ImagePathItem and etc?

Reply:
It doesn't have to be special, it just maps to a separate type that KBFS stores.

Reply:
i'd make these types from a UI perspective and not any other system. aka we avoid using core types

Reply:
Sure, I'm fine with that. I'll accept your PR and then make a few cosmetic changes.